command/git.go

```diff
@@ -51,34 +54,82 @@ func (c *gitFetchProject) ParseParams(params map[string]interface{}) error {
 	return nil
 }
 
-// Execute gets the source code required by the project
-func (c *gitFetchProject) Execute(ctx context.Context,
-	comm client.Communicator, logger client.LoggerProducer, conf *model.TaskConfig) error {
+func (c *gitFetchProject) buildHTTPCloneCommand(projectRef *model.ProjectRef) ([]string, error) {
+	location, err := projectRef.HTTPLocation()
+	if err != nil {
+		return nil, err
+	}
+	location.User = url.UserPassword(c.GithubOauthToken, "x-oauth-basic")
```

Comment:
given that we take the string form of this *and* run the command with xtrace?

Reply:
I turned off xtrace for this one command and printed a redacted version of the pull command without token.